File: 백준/Silver/2164. 카드2/카드2.cc
#include <iostream>
#include <queue>

using namespace std;

int main() {
    int N;
    cin >> N;

    queue<int> Q;
    for (int i = 1; i <= N; i++)
    {
        Q.push(i);
    }

    int i = 0;
    while(Q.size() > 1)
    {
        if (i%2 == 0) {
            Q.pop();
        } else {
            int temp = Q.front();
            Q.pop();
            Q.push(temp);
        }
        i++;
    }
    
    cout << Q.front() << endl;
    return 0;
}

File: 백준/Silver/2164. 카드2/카드2.py
import sys
from collections import deque
input = sys.stdin.readline

N = int(input())

queue = deque([i+1 for i in range(N)])

while(len(queue) > 1):
  queue.popleft()
  if len(queue) == 1: break

  temp = queue.popleft()
  queue.append(temp)

print(queue.pop())